packages/components/src/components/StepIndicator/StepIndicatorSegment.tsx

```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import { VisuallyHidden } from "ariakit/visually-hidden";
+import { Box } from "../../primitives/Box";
+
+export interface StepIndicatorSegmentProps
+  extends Omit<React.HTMLAttributes<HTMLOListElement>, "color"> {
+  /** The accessible text content of the step. */
+  children: string;
+  /** The current step that shows the user’s position. Inherited from StepIndicator. */
+  current?: number;
+  /** The index within the array of segments. Inherited from StepIndicator. */
+  index?: number;
```

Comment:
Nit: Should step indicator just take in a prop called `isActiveStep` or something instead of having to compare current and index inside this component and let the parent component handle the logic of what makes it active.

Reply:
@RohitRajendran what would `isActiveStep` do? Is it a `boolean`? How is that different from using `current`? We'd still have to somehow move `isActiveStep` through the child components, right? Maybe I'm not following.

Reply:
Oh wait. I think I get it now.

Reply:
@RohitRajendran take a look now and let me know if thats on track.

Reply:
👌